cmd/initialize_with_genesis.cpp

```diff
@@ -0,0 +1,335 @@
+/*
+   Copyright 2021 The Silkworm Authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+#include <stdlib.h>
+
+#include <fstream>
+#include <filesystem>
+#include <iostream>
+
+#include <CLI/CLI.hpp>
+#include <boost/endian/conversion.hpp>
+#include <nlohmann/json.hpp>
+
+#include <silkworm/common/base.hpp>
+#include <silkworm/common/chain_genesis.hpp>
+#include <silkworm/common/log.hpp>
+#include <silkworm/common/util.hpp>
+#include <silkworm/db/access_layer.hpp>
+#include <silkworm/db/stages.hpp>
+#include <silkworm/db/tables.hpp>
+#include <silkworm/etl/collector.hpp>
+#include <silkworm/trie/hash_builder.hpp>
+#include <silkworm/types/account.hpp>
+
+using namespace silkworm;
+
+constexpr uint8_t genesis_body[] = {195, 128, 128, 192};
+constexpr uint8_t genesis_receipts[] = {246};
+constexpr evmc::bytes32 kNullOmmers{0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347_bytes32};
+
+constexpr size_t migrations_count = 20;
+std::string last_header_key = "LastHeader";
+std::pair<Bytes, Bytes> init_migrations_value[migrations_count] = {
+    {*from_hex("6163635f6368616e67655f7365745f6475705f736f72745f3138"),
```

Comment:
1. IMHO for better readability keys should be strings with values like "acc_change_set_dup_sort_18". When inserting, strings can be casted to `ByteView` with `byte_view_of_c_str` (in core/silkworm/common/util.hpp)
2. I wouldn't use migration values at all here. They only serve a debug purpose, don't they? If we need a value, we could use some dummy like `*from_hex("a0")` for everything.

So I'd have the following code:
```
constexpr const char* kMigration[] = {
    "stages_to_use_named_keys",
    "unwind_stages_to_use_named_keys",
    "stagedsync_to_use_stage_blockhashes",
    ...
}

Reply:
Totally agree on this but I extend the question.
Given is very unlikely we(or anybody) use Silkworm tool to instantiate a new db do we really need to take into account migrations ? Afaik when in beta db _schema_ will be consolidated and I expect a check only on a db var.

Reply:
I mean ... as long as Silkworm is in catch-up phase with TG I don't think is worth the effort to take into account migrations.
cc @AlexeyAkhunov

Reply:
> Totally agree on this but I extend the question.
> Given is very unlikely we(or anybody) use Silkworm tool to instantiate a new db do we really need to take into account migrations ? Afaik when in beta db _schema_ will be consolidated and I expect a check only on a db var.

Right, that's a good point. Perhaps we can remove migrations from this PR altogether and revisit the matter later.